src/classes/PMT_Payment_TEST.cls

```diff
@@ -381,6 +381,234 @@ private class PMT_Payment_TEST {
+        npe01__OppPayment__c invalidPaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = true);
+        insert invalidPaidPayment;
+
+        invalidPaidPayment = queryForPaymentForTesting(invalidPaidPayment.Id);
+
+        System.assertEquals(true, invalidPaidPayment.npe01__Paid__c, 'Paid should be true, but it is actually: ' + invalidPaidPayment.npe01__Paid__c);
+        System.assertEquals(null, invalidPaidPayment.npe01__Payment_Date__c, 'The payment date should be today, but it is actually: ' + invalidPaidPayment.npe01__Payment_Date__c);
+
+        // Test the invalid unpaid insertion case
+        npe01__OppPayment__c invalidUnpaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = false);
+        insert invalidUnpaidPayment;
+
+        invalidUnpaidPayment = queryForPaymentForTesting(invalidUnpaidPayment.Id);
+
+        System.assertEquals(false, invalidUnpaidPayment.npe01__Paid__c, 'Paid should be false, but it is actually: ' + invalidUnpaidPayment.npe01__Paid__c);
+        System.assertEquals(null, invalidUnpaidPayment.npe01__Scheduled_Date__c, 'The payment date should be null, but it is actually: ' + invalidUnpaidPayment.npe01__Scheduled_Date__c);
+
+        // Test the invalid written off insertion case
+        npe01__OppPayment__c invalidWrittenOffPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Written_Off__c = true);
+        insert invalidWrittenOffPayment;
+
+        invalidWrittenOffPayment = queryForPaymentForTesting(invalidWrittenOffPayment.Id);
+
+        System.assertEquals(true, invalidWrittenOffPayment.npe01__Written_Off__c, 'Paid should be true, but it is actually: ' + invalidWrittenOffPayment.npe01__Written_Off__c);
+        System.assertEquals(null, invalidWrittenOffPayment.npe01__Payment_Date__c, 'The payment date should be today, but it is actually: ' + invalidWrittenOffPayment.npe01__Payment_Date__c);
+
+    }
+
+    /**
+     * @description Helper function to help set the payment attributes as necessary for testing that accounting data consistency is enforced. 
+     */
+    private static npe01__OppPayment__c updatePaymentForTesting(npe01__OppPayment__c payment, Boolean isPaid, Boolean isWrittenOff, Boolean hasPaymentDate, Boolean hasScheduledDate) {
+        payment.npe01__Paid__c = isPaid;
+        payment.npe01__Written_Off__c = isWrittenOff;
+
+        if (hasPaymentDate == true) {
+            payment.npe01__Payment_Date__c = Date.today();
+        } else {
+            payment.npe01__Payment_Date__c = null;
+        }
+
+        if (hasScheduledDate == true) {
+            payment.npe01__Scheduled_Date__c = Date.today();
+        } else {
+            payment.npe01__Scheduled_Date__c = null;
+        }
+
+        return payment;
+    }
+
+    /**
+     * @description Helper function to query for payments given a specific ID. 
+     */
+    private static npe01__OppPayment__c queryForPaymentForTesting(Id paymentId) {
```

Comment:
Being a unit test helper method is there *ever* a scenario where an invalid payment record Id would be passed to this method? I can't think of one, so I think this can be rewritten as 
```
return [SELECT Id, npe01__Paid__c, npe01__Written_Off__c, npe01__Payment_Date__c, npe01__Scheduled_Date__c 
    FROM npe01__OppPayment__c
    WHERE Id =: paymentId];
```

Reply:
I refactored this to be like the other method in here instead, in case we need to check the currencycode, etc. via this method at a later point.